index.js

```diff
@@ -17,12 +18,43 @@ var userAgent = 'Mozilla/5.0 AppleWebKit/537.36 Chrome/50.0.2653.0 Safari/537.36
 follow(process.argv[2], start)
 
 function setCookie (url, headers) {
-  var cookieHeader = headers['set-cookie']
-  if (!cookieHeader) return 0
-  cookieHeader.forEach(function (value) {
+  if (!headers['set-cookie']) return 0
+  headers['set-cookie'].forEach(function (value) {
     cookies.store(url, value)
   })
-  return cookieHeader.length
+  return headers['set-cookie'].length
+}
+
+function setProtocol (res) {
+  res.protocol = 'HTTP'
+  if (res.socket._spdyState) {
+    var spdyState = res.socket._spdyState.parent
+    switch (spdyState.alpnProtocol || spdyState.npnProtocol) {
+      case 'h2':
+        res.httpVersion = '2.0'
+        res.httpVersionMajor = '2'
+        res.httpVersionMinor = '0'
+        break
+      case 'spdy/3.1':
+        res.protocol = 'SPDY'
+        res.httpVersion = '3.1'
+        res.httpVersionMajor = '3'
+        res.httpVersionMinor = '1'
+        break
+      case 'spdy/3':
+        res.protocol = 'SPDY'
+        res.httpVersion = '3.0'
+        res.httpVersionMajor = '3'
+        res.httpVersionMinor = '0'
+        break
+      case 'spdy/2':
+        res.protocol = 'SPDY'
+        res.httpVersion = '2.0'
+        res.httpVersionMajor = '2'
+        res.httpVersionMinor = '0'
+        break
```

Comment:
@jhermsmeier I'm not a SPDY expert, but is there a chance that there will come new versions  of the SPDY protocol often that we should support? I'm just thinking if it would make sense to extract the version number our of the SPDY protocol string or if we should just leave it hardcoded for now?